src/snowflake/connector/cache.py

```diff
@@ -219,7 +236,8 @@ def _update(
                 g = iter(other.items())
             to_insert = {k: CacheEntry(expiry=expiry, entry=v) for k, v in g}
         elif isinstance(other, SFDictCache):
-            other._clear_expired_entries()
+            with other._lock:
+                other._clear_expired_entries()
```

Comment:
just call the public api shall be good?
```suggestion
            other.clear_expired_entries()
```